File: src/beginner_tutorials/src/iot_talker.cpp
#include "ros/ros.h"
#include "beginner_tutorials/IoTSensor.h"

int main(int argc, char **argv){
    ros::init(argc, argv, "iot_talker");        
    ros::NodeHandle n;
    ros::Publisher iot_publisher = n.advertise<beginner_tutorials::IoTSensor>("iot_line",1000);
    ros::Rate loop_rate(2);
    beginner_tutorials::IoTSensor iot_sensor;
    while (ros::ok()){
        iot_sensor.id = 457;
        iot_sensor.name = "Indor 2";
        iot_sensor.temperature = 23.20;
        iot_sensor.humidity = 43.99;
        iot_publisher.publish(iot_sensor);
        ROS_INFO("Message sent!!! %s", iot_sensor.name.c_str());
        ros::spinOnce();
        loop_rate.sleep();
    }
    return 0;
}

File: src/beginner_tutorials/src/scripts/iot_talker.py
#!/usr/bin/env python
import rospy
from beginner_tutorials.msg import IoTSensor
from time import sleep

def iot_talker():
    publisher = rospy.Publisher("iot_line", IoTSensor, queue_size=10)
    rospy.init_node("iot_sensor")
    sensor = IoTSensor()
    while not rospy.is_shutdown():
        sensor.id = 123
        sensor.name = "Outdor 1"
        sensor.temperature = 20.23
        sensor.humidity = 99.43
        publisher.publish(sensor)
        rospy.loginfo("IoTSensor: \n%s"%sensor)
        sleep(.5)

if __name__ == "__main__":
    try:
        iot_talker()
    except rospy.ROSInterruptException:
        pass
